modules/apps/frontend-js/frontend-js-walkthrough-web/test/js/Walkthrough.test.js

```diff
@@ -0,0 +1,251 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import '@testing-library/jest-dom/extend-expect';
+import {cleanup, render, screen} from '@testing-library/react';
+import userEvents from '@testing-library/user-event';
+import {navigate} from 'frontend-js-web';
+import React from 'react';
+
+import Walkthrough from '../../src/main/resources/META-INF/resources/Walkthrough';
+import {
+	BOX_SHADOW_ELEMENT_MOCK,
+	DOM_STRUCTURE_FOR_PLACING_STEPS,
+	INVALID_NODE_SELECTOR_MOCK,
+	MULTI_PAGES_MOCK,
+	PAGE_MOCK,
+	PAGE_WITH_PREVIOUS_MOCK,
+} from './__fixtures__/walkthroughMock';
+
+function setupDocument() {
+	const domStructureReference = document.createElement('div');
+
+	domStructureReference.innerHTML = DOM_STRUCTURE_FOR_PLACING_STEPS;
+
+	domStructureReference.id = 'test-structure';
+
+	document.body.appendChild(domStructureReference);
+
+	return () => {
+		window.localStorage.clear();
+
+		const nodeToBeRemoved = document.getElementById('test-structure');
+
+		document.body.removeChild(nodeToBeRemoved);
+
+		nodeToBeRemoved.remove();
+	};
+}
+
+function renderWalkthrough(props) {
+	return render(<Walkthrough {...props} />, {
+		baseElement: document.getElementById('app_root'),
+	});
+}
+
+jest.mock('frontend-js-web', () => ({
+	navigate: jest.fn((url) => url),
+}));
+
+const USER_ID = 42;
+
+describe('Walkthrough', () => {
+	window.themeDisplay = {
+		getLayoutRelativeURL: jest.fn(() => '/home'),
+		getUserId: jest.fn(() => USER_ID),
+	};
+
+	afterEach(() => {
+		cleanup();
+	});
+
+	beforeEach(() => {
+		jest.restoreAllMocks();
+	});
+
+	it('renders', () => {
+		const cleanUp = setupDocument();
+
+		const {container} = renderWalkthrough(PAGE_MOCK);
+
+		expect(container).toBeInTheDocument();
+
+		cleanUp();
```

Comment:
Could this stay on `afterEach` instead of calling on every test?

Reply:
This cleanUp comes from the return of setupDocument. I can't get the reference inside the afterEach/beforeEach

Reply:
I created a reference on described scope called `cleanUpDocument`, and I'm assigning the cleanUp function from beforeEach, and it's working.

![image](https://user-images.githubusercontent.com/7663513/187766245-01907cf3-5eab-4a06-868c-f53acecef8a3.png)